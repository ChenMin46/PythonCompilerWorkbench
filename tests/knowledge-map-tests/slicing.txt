x[1:5]
x[1:]
x[:5]
x[a:b]
x[a:]
x[:b]
x[:]
x[a:b:c]
part = whole[1:5]
