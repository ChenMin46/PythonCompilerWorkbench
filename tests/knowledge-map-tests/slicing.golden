x[1:5]
x[1:]
x[:5]
x[a:b]
x[a:]
x[:b]
x[:]
x[a:b:c]
part = whole[1:5]
===
Module
  body: [
    Expr
      value: Subscript >> 'x[1:5]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> '1:5'
          lower: Num >> '1'
            n: 1
          upper: Num >> '5'
            n: 5
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[1:]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> '1:'
          lower: Num >> '1'
            n: 1
          upper: None
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[:5]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> ':5'
          lower: None
          upper: Num >> '5'
            n: 5
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[a:b]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> 'a:b'
          lower: Name >> 'a'
            id: 'a'
            ctx: Load
          upper: Name >> 'b'
            id: 'b'
            ctx: Load
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[a:]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> 'a:'
          lower: Name >> 'a'
            id: 'a'
            ctx: Load
          upper: None
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[:b]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> ':b'
          lower: None
          upper: Name >> 'b'
            id: 'b'
            ctx: Load
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[:]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice
          lower: None
          upper: None
          step: None
        ctx: Load
    Expr
      value: Subscript >> 'x[a:b:c]'
        value: Name >> 'x'
          id: 'x'
          ctx: Load
        slice: Slice >> 'a:b:c'
          lower: Name >> 'a'
            id: 'a'
            ctx: Load
          upper: Name >> 'b'
            id: 'b'
            ctx: Load
          step: Name >> 'c'
            id: 'c'
            ctx: Load
        ctx: Load
    Assign >> 'part = whole[1:5]'
      targets: [
        Name >> 'part'
          id: 'part'
          ctx: Store
      ]
      value: Subscript >> 'whole[1:5]'
        value: Name >> 'whole'
          id: 'whole'
          ctx: Load
        slice: Slice >> '1:5'
          lower: Num >> '1'
            n: 1
          upper: Num >> '5'
            n: 5
          step: None
        ctx: Load
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "1"
                    }, 
                    {
                      "id": "tid_1", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "5"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "1"
                    }, 
                    {
                      "id": "tid_2", 
                      "type": "token", 
                      "value": ":"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "tid_3", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "5"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "a"
                    }, 
                    {
                      "id": "tid_4", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "b"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "a"
                    }, 
                    {
                      "id": "tid_5", 
                      "type": "token", 
                      "value": ":"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "tid_6", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "b"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[:]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "a"
                    }, 
                    {
                      "id": "tid_7", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "b"
                    }, 
                    {
                      "id": "tid_8", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "type": "Name", 
                      "value": "c"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "ctx": "Store", 
              "id": "id_0", 
              "type": "Name", 
              "value": "part"
            }, 
            " ", 
            {
              "id": "tid_9", 
              "type": "token", 
              "value": "="
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "type": "Name", 
                  "value": "whole"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "1"
                    }, 
                    {
                      "id": "tid_10", 
                      "type": "token", 
                      "value": ":"
                    }, 
                    {
                      "id": "id_0", 
                      "type": "Num", 
                      "value": "5"
                    }
                  ], 
                  "id": "id_0", 
                  "type": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Load", 
              "id": "id_0", 
              "type": "Subscript"
            }
          ], 
          "id": "id_0", 
          "type": "Assign"
        }
      ], 
      "id": "id_0", 
      "type": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "type": "ROOT_NODE"
}
